### PR TITLE
Edit docs for config templating

### PR DESCRIPTION
**Pull request by boraarslan**

### CODE REVIEW

**Comment by boraarslan** [comment]
Should we change the tutorial to use only the templating for config? I only added it as an alternative for now, but I believe it is more ergonomic to use templating instead of writing from stdout.

File: docs/configuration/node-config.md [hunk 1]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
 | fast_field_cache_capacity | Fast field cache capacity on a Searcher. | 10G |
 | split_footer_cache_capacity | Split footer cache (it is essentially the hotcache) capacity on a Searcher. | 1G |
 | max_num_concurrent_split_streams | Maximum number of concurrent split stream requests running on a Searcher. | 100 |
+
+## Templating
```
Comment (guilload) [inline docs/configuration/node-config.md:48]:
```suggestion
## Using environment variables in the configuration
```

File: docs/configuration/node-config.md [hunk 2]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
...
+
+It is possible to pass environment variables to config:
```
Comment (guilload) [inline docs/configuration/node-config.md:50]:
```suggestion
You can use environment variable references in the config file to set values that need to be configurable during deployment. To do this, use:

${VAR_NAME}

Where  `VAR_NAME` is the name of the environment variable.

Each variable reference is replaced at startup by the value of the environment variable. The replacement is case-sensitive and occurs before the configuration file is parsed. References to undefined variables throw an error unless you specify a default value or custom error text.

To specify a default value, use:

${VAR_NAME:-default_value}

Where `default_value` is the value to use if the environment variable is not set.
```

File: docs/configuration/node-config.md [hunk 3]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
...
+```yaml
+# config.yaml
+version: 0
+cluster_id: quickwit-cluster
+node_id: my-unique-node-id
+listen_address: ${QW_LISTEN_ADDRESS}
+rest_listen_port: ${QW_LISTEN_PORT:-1111}
+```
+
+Will get interpreted by Quickwit as:
```
Comment (guilload) [inline docs/configuration/node-config.md:73]:
```suggestion
Will be interpreted by Quickwit as:
```

File: docs/configuration/node-config.md [hunk 4]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
...
+
+```yaml
+version: 0
+cluster_id: quickwit-cluster
+node_id: my-unique-node-id
+listen_address: 0.0.0.0
+rest_listen_port: 4444 #< 1111 if QW_LISTEN_PORT is not set
```
Comment (guilload) [inline docs/configuration/node-config.md:80]: Let's just not export `QW_LISTEN_PORT` and render with the default value `1111`. There's already an example with `QW_LISTEN_ADDRESS`

File: docs/configuration/node-config.md [hunk 5]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
 | fast_field_cache_capacity | Fast field cache capacity on a Searcher. | 10G |
 | split_footer_cache_capacity | Split footer cache (it is essentially the hotcache) capacity on a Searcher. | 1G |
 | max_num_concurrent_split_streams | Maximum number of concurrent split stream requests running on a Searcher. | 100 |
+
+## Templating
+
+It is possible to pass environment variables to config:
+```
+<config_field>: ${<ENV_VAR_NAME>}
```
Comment (guilload) [inline docs/configuration/node-config.md:52]:
```suggestion
<config_field>: ${VAR_NAME}
```

File: docs/configuration/node-config.md [hunk 6]
```diff
@@ -44,3 +44,38 @@ This section contains the configuration options for a Searcher.
...
+or
+<config_field>: ${<ENV_VAR_NAME>:-<default value>}
```
Comment (guilload) [inline docs/configuration/node-config.md:54]:
```suggestion
<config_field>: ${VAR_NAME:-default_value}
```

File: docs/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md [hunk 7]
```diff
@@ -95,6 +95,14 @@ default_index_root_uri: ${S3_PATH}
 " > config.yaml
 ```
 
+> You can also pass environment variables directly:
```
Comment (guilload) [inline docs/tutorials/tutorial-hdfs-logs-distributed-search-aws-s3.md:98]: I think this tutorial deserves some rewriting. I'll take care of it before the next release. You can dismiss those changes for now.

**Review by guilload — CHANGES_REQUESTED** [review]

**Review by guilload — APPROVED** [review]

**Pull request merged**